### PR TITLE
Always return the root of the merkle tree

### PR DESCRIPTION
**Pull request by ashamis**

### CODE REVIEW

**Comment by jumaffre** [comment]
Not sure I understand why these changes are necessary. Could the existing `add_result()` also be called when `data` is empty (and not append anything to the Merkle Tree)?

File: src/node/rpc/frontend.h [hunk 1]
```diff
@@ -622,6 +625,11 @@ namespace ccf
 
       history->clear_on_result();
 
+      if (!has_updated_merkle_root)
```
Comment (olgavrou) [inline src/node/rpc/frontend.h]: when would this not have been updated?

**Comment by olgavrou** [comment]
frontend.h has unit tests in place. We need to start unit testing for any new pbft additions to the frontend.h code in those unit tests.

File: src/node/rpc/frontend.h (hunk 1 above)
Comment (ashamis) [reply src/node/rpc/frontend.h]: if there is no read or write set in the transaction

**Pull request merged**